base/strings/types.jl

```diff
@@ -1,6 +1,6 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-# SubString and RevString types
+# SubString type
```

Comment:
This file is kind of inconsistent now. Maybe better rename it to substring.jl and move unrelated functions elsewhere?

Reply:
Sounds good. Once I'm done with implementations and cleanup here I'll make that a separate commit.